Library/Homebrew/utils/ruby.sh

```diff
@@ -1,8 +1,15 @@
-test-ruby () {
-  [[ ! -x $1 ]] && { echo "false"; return 1; }
-  "$1" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e \
+test_ruby () {
+  if [[ ! -x $1 ]]
+  then
+    return 1
+  fi
+
+  local ruby_status
+  ruby_status=$("$1" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e \
     "puts Gem::Version.new(RUBY_VERSION.to_s.dup).to_s.split('.').first(2) == \
```

Comment:
Shall we make Ruby return the exit status instead of capturing the output and comparing to "true"?
Something like this
```sh
"$1" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt -rrubygems -e \
   "exit(1) unless Gem::Version.new(RUBY_VERSION.to_s.dup).to_s.split('.').first(2) == \
                   Gem::Version.new('$required_ruby_version').to_s.split('.').first(2)" 2>/dev/null
```

Reply:
Works for me. Can use `abort` here and would prefer `if ! &&` over `unless &&`

Reply:
Implemented `abort` paired with `!=` instead of `! .... ==`